File: SlidingWindow/findBestSubarray.js
function findBestSubarray(nums, k) {
    let curr = 0;
    let answer = 0;

    for (let i = 0; i < k; i++) {
        curr += nums[i];
    }

    answer = curr;

    for (let i = k; i < nums.length; i++) {
        curr += nums[i];
        curr -= nums[i - k];
        answer = Math.max(answer, curr);
    }

    return answer;
}

console.log(findBestSubarray([3, -1, 4, 12, -8, 5, 6], 4))

File: Arrays&Strings/SlidingWindow/findBestSubarray.py
def find_best_subarray(nums: list[int], k: int) -> int:
    curr, answer = 0, 0

    for i in range(k):
        curr += nums[i]

    answer = curr

    for i in range(k, len(nums)):
        curr += nums[i]
        curr -= nums[i - k]
        answer = max(answer, curr)

    return answer

print(find_best_subarray([3, -1, 4, 12, -8, 5, 6], 4))